conn_test.go

```diff
@@ -656,41 +656,56 @@ func TestNonblockingHandshakeAndDataFlow(t *testing.T) {
 	// Send ClientHello
 	clientAlert = client.Handshake()
 	assertEquals(t, clientAlert, AlertNoAlert)
+	assertEquals(t, client.GetHsState(), StateClientWaitSH)
 	serverAlert = server.Handshake()
 	assertEquals(t, serverAlert, AlertWouldBlock)
+	assertEquals(t, server.GetHsState(), StateServerStart)
 
 	// Release ClientHello
 	cbConn.Flush()
 
 	// Process ClientHello, send server first flight.
-	for {
-		serverAlert = server.Handshake()
-		if serverAlert == AlertWouldBlock {
-			break
-		}
-		assertEquals(t, serverAlert, AlertNoAlert)
-	}
+	serverAlert = server.Handshake()
```

Comment:
Might be more elegant to do this as a list + for-loop.

Reply:
@bifurcation: Done.